android/app/src/main/java/org/uwaterloo/subletr/pages/createaccount/CreateAccountPageView.kt

```diff
@@ -1,13 +1,467 @@
+					)
+					Row {
+						Text(
+							text = stringResource(id = R.string.app_name_black_part),
+							style = MaterialTheme.typography.titleLarge,
+						)
+						Text(
+							text = stringResource(id = R.string.app_name_pink_part),
+							style = MaterialTheme.typography.titleLarge,
+							color = subletrPink,
+						)
+					}
+				}
+
+				Spacer(
+					modifier = Modifier.weight(weight = 2.0f)
+				)
+			}
+
+			Spacer(
+				modifier = Modifier.weight(weight = 2.0f)
+			)
+
+			TextField(
+				modifier = Modifier
+					.fillMaxWidth(ELEMENT_WIDTH),
+				singleLine = true,
+				shape = RoundedCornerShape(size = 100.dp),
+				colors = TextFieldDefaults.colors(
+					unfocusedContainerColor = buttonBackgroundColor,
+					focusedContainerColor = buttonBackgroundColor,
+					unfocusedIndicatorColor = Color.Transparent,
+					focusedIndicatorColor = Color.Transparent,
+				),
+				placeholder = {
+					Text(
+						text = stringResource(id = R.string.first_name),
+						color = secondaryTextColor,
+					)
+				},
+				value = uiState.firstName,
+				onValueChange = {
+					viewModel.updateUiState(
+						CreateAccountPageUiState.NewAccountInfo(
+							firstName = it,
+							lastName = uiState.lastName,
+							email = uiState.email,
+							password = uiState.password,
+							confirmPassword = uiState.confirmPassword,
+							gender = uiState.gender,
+						)
+					)
+				}
+			)
+
+			Spacer(
+				modifier = Modifier.weight(weight = 2.0f)
+			)
+
+			TextField(
+				modifier = Modifier
+					.fillMaxWidth(ELEMENT_WIDTH),
+				singleLine = true,
+				shape = RoundedCornerShape(size = 100.dp),
+				colors = TextFieldDefaults.colors(
+					unfocusedContainerColor = buttonBackgroundColor,
+					focusedContainerColor = buttonBackgroundColor,
+					unfocusedIndicatorColor = Color.Transparent,
+					focusedIndicatorColor = Color.Transparent,
+				),
+				placeholder = {
+					Text(
+						text = stringResource(id = R.string.last_name),
+						color = secondaryTextColor,
+					)
+				},
+				value = uiState.lastName,
+				onValueChange = {
+					viewModel.updateUiState(
+						CreateAccountPageUiState.NewAccountInfo(
+							firstName = uiState.firstName,
+							lastName = it,
+							email = uiState.email,
+							password = uiState.password,
+							confirmPassword = uiState.confirmPassword,
+							gender = uiState.gender,
+						)
+					)
+				}
+			)
+
+			Spacer(
+				modifier = Modifier.weight(weight = 2.0f)
+			)
+
+			TextField(
+				modifier = Modifier
+					.fillMaxWidth(ELEMENT_WIDTH),
+				singleLine = true,
+				shape = RoundedCornerShape(size = 100.dp),
+				colors = TextFieldDefaults.colors(
+					unfocusedContainerColor = buttonBackgroundColor,
+					focusedContainerColor = buttonBackgroundColor,
+					unfocusedIndicatorColor = Color.Transparent,
+					focusedIndicatorColor = Color.Transparent,
+				),
+				keyboardOptions = KeyboardOptions(
+					keyboardType = KeyboardType.Email,
+					autoCorrect = false
```

Comment:
Oof, trailing comma lol

Reply:
from what I saw in the code I thought they were good oops

Reply:
oh there SHOULD be a trailing comma, I'll add it